Makefile

```diff
@@ -149,22 +149,23 @@ endif
 
 # Qt5 dirs (we can't indent with tabs here)
 ifneq ($(detected_OS),Windows)
- QT5_PCFILEDIR := $(shell pkg-config --variable=pcfiledir Qt5Core 2>/dev/null)
- QT5_LIBDIR := $(shell pkg-config --variable=libdir Qt5Core 2>/dev/null)
- ifeq ($(QT5_PCFILEDIR),)
-  QT5_PCFILEDIR := $(QTDIR)/lib/pkgconfig
-  QT5_LIBDIR := $(QTDIR)/lib
-  # some manually installed Qt5 instances have wrong paths in their *.pc files, so we pass the right one to the linker here
-  ifeq ($(detected_OS),Darwin)
-   NIM_PARAMS += -L:"-framework Foundation -framework AppKit -framework Security -framework IOKit -framework CoreServices -framework LocalAuthentication"
-   # Fix for failures due to 'can't allocate code signature data for'
-   NIM_PARAMS += --passL:"-headerpad_max_install_names"
-   NIM_PARAMS += --passL:"-F$(QT5_LIBDIR)"
-
+ # System QT installation might mess the build. Since QTDIR must always be exported,
+ # $QTDIR/lib/pkgconfig must always be there unless Qt6, so lets use it by default.
+ # some manually installed Qt5 instances have wrong paths in their *.pc files, so we pass the right one to the linker here
```

Comment:
Does this comment really need 3 lines? You can say this much more succintly:
```Makefile
 # Depend explicitly on QTDIR because system QT can have wrong paths for *.pc files. 
```
Long comments are almost never read by anyone.

Reply:
Plenty of broken distros around :D.

Reply:
Then you should query the desired `qmake` for its paths